code/modules/mob/living/carbon/human/species.dm

```diff
@@ -610,7 +610,11 @@
 
 	var/num_arms = H.get_num_arms()
 	var/num_legs = H.get_num_legs()
-
+	if(istype(I, /obj/item/clothing))
+		var/obj/item/clothing/C = I
+		if(C.dwarf_only)
```

Comment:
I already mentioned this to goof. @RemieRichards some clothes are compatible for dwarves to wear (leather explorer satchels etc) but most have height distortions or width issues, so some testing will be needed in the case of the suggested white/black-lists.

A similar system will probably be used for other races with non-standard body sprite, so this is good.